airflow/migrations/versions/0129_2_8_0_add_clear_number_to_dag_run.py

```diff
@@ -43,7 +43,7 @@ def upgrade():
             sa.Column(
                 "clear_number",
                 sa.Integer,
-                default=0,
+                server_default="0",
```

Comment:
I think we need to handle MSSQL default in the downgrade if we go with server_default

Reply:
The SQL Query generated by this is `ALTER TABLE dag_run ADD COLUMN clear_number INTEGER NOT NULL DEFAULT 0;`. 

So I think a simple drop should work. 

Any resources where I can read more about what you are referring for MSSQL?

Reply:
Nevermind, I've found answer in comments

Reply:
I've had to spend some time figuring out how to make this work with MSSQL, and I think this is an approach that 'works':

https://github.com/apache/airflow/pull/34344/files - the PR is currently failing because of failing ER diagram and black pre-commit checks, but the Migration CI seems to be working. I don't have access to my laptop right now, but will be able to fix the pre-commit check later this evening... or @abhishekbhakat feel free to adopt the logic into this PR to merge it in, if you are able to get to it faster than I am 👍 

Other seemingly more 'correct' ways of handling the migration for MSSQL were failing.

For example, using the following drop_column command still fails for MSSQL: 

`batch_op.drop_column("clear_number", mssql_drop_default=True)`

```
sqlalchemy.exc.ProgrammingError: (pyodbc.ProgrammingError) ('42000', "[42000] [Microsoft][ODBC Driver 18 for SQL Server][SQL Server]Incorrect syntax near 'constraint'. (102) (SQLExecDirectW)")
[SQL: declare @const_name varchar(256)
select @const_name = QUOTENAME([name]) from sys.default_constraints
where parent_object_id = object_id('dag_run')
and col_name(parent_object_id, parent_column_id) = 'clear_number'
exec('alter table dag_run drop constraint ' + @const_name)]
```

Reply:
Yeah! ER Diagram is failing.